subprojects/maven/src/main/java/org/gradle/api/publication/maven/internal/action/AbstractMavenPublishAction.java

```diff
@@ -71,18 +70,9 @@ protected AbstractMavenPublishAction(File pomFile, File metadataFile, List<File>
 
         Model pom = parsePom(pomFile);
```

Comment:
⭕️ I'm a bit confused about this, why do we have special handling for the POM here?

Reply:
Doesn't make any sense to me either. I'll try to refactor it when this PR is merged.